source/slang/slang-type-layout.h

```diff
@@ -753,6 +753,9 @@ class ProgramLayout : public ScopeLayout
         /// arguments that have been used to specialize the program.
         ///
     List<RefPtr<TypeLayout>> taggedUnionTypeLayouts;
+
+        /// Holds all of the string literals that have been hashed
+    StringSlicePool hashedStringLiteralPool;
```

Comment:
Okay, so this is where the interaction with `StringSlicePool` in the IR step is coming in, as well as where the API contract that doesn't guarantee null termination comes from (I assume).